### PR TITLE
[INLONG-2781][DataProxy] Update netty version to 4.x

### LINKED ISSUE #2781: [Feature][DataProxy] Update netty version to 4.x
### Description

update netty version to 4 in inlongTcp*  file of dataproxy

### Use case

update netty version to 4 in inlongTcp*  file of dataproxy

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by baomingyu**
update netty version to 4 in inlongTcp* file of dataproxy 

Fixes #2781

### CODE REVIEW

**Review by healchow — APPROVED** [review]

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelHandler.java [hunk 1]
```diff
@@ -223,41 +208,59 @@ private void responsePackage(ResultCode code, MessageEvent e)
...
-        LOG.error("exception caught", e.getCause());
-        super.exceptionCaught(ctx, e);
-        if (e.getChannel() != null) {
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) {
+        LOG.error("exception caught cause = {}", cause);
+        if (ctx.channel() != null) {
             try {
-                e.getChannel().disconnect();
-                e.getChannel().close();
+                ctx.fireExceptionCaught(cause);
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelHandler.java:219]: If ctx.fireExceptionCaught get an Exception, the operation of disconnect and close will not be invoked.

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelHandler.java [hunk 2]
```diff
@@ -191,28 +175,29 @@ private void addMetric(boolean result, long size, Event event) {
...
         builder.setResult(code);
 
         // encode
         byte[] responseBytes = builder.build().toByteArray();
         //
-        ChannelBuffer buffer = ChannelBuffers.wrappedBuffer(responseBytes);
-
-        Channel remoteChannel = e.getChannel();
+        ByteBuf buffer = ByteBufAllocator.DEFAULT.buffer(responseBytes.length);
+        buffer.writeBytes(responseBytes);
```
Comment (luchunliang) [inline inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelHandler.java:192]: It is better to use Unpooled.wrappedBuffer(byte[]).

**Review by luchunliang — APPROVED** [review]

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelHandler.java (hunk 1 above)
Comment (baomingyu) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelHandler.java]: done

File: inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelHandler.java (hunk 2 above)
Comment (baomingyu) [reply inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpChannelHandler.java]: done

**Pull request merged**